### ISSUE TITLE
Foreign-key validation throws an exception when the reference table has missing or extra headers

### ISSUE THREAD

**Issue by didiez** [frictionlessdata/goodtables-py#347]
# Overview

The exception is thrown when validating two csv files with foreign keys between them.
The schema allows extra headers and missing non-required headers, so the csv could have less (or more) columns than the defined schema.

```python
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    report = validate("datapackage.json", checks=["structure", "schema", "foreign-key"], order_fields=True, infer_fields=False)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/goodtables/validate.py", line 80, in validate
    report = inspector.inspect(source, **options)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/goodtables/inspector.py", line 82, in inspect
    table_warnings, table_report = task.get()
  File "/home/didiez/anaconda3/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
  File "/home/didiez/anaconda3/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/goodtables/inspector.py", line 200, in __inspect_table
    success = prepare_func(stream, schema, extra)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/goodtables/contrib/checks/foreign_key.py", line 48, in prepare
    current_resource_name=extra['resource-name'])
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/goodtables/contrib/checks/foreign_key.py", line 116, in _get_relations
    relations[resource_name] = resource.read(keyed=True)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/datapackage/resource.py", line 377, in read
    foreign_keys_values=foreign_keys_values, **options)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/tableschema/table.py", line 353, in read
    for count, row in enumerate(rows, start=1):
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/tableschema/table.py", line 215, in iter
    for row_number, headers, row in iterator:
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/tableschema/table.py", line 509, in builtin_processor
    row, row_number=row_number, exc_handler=exc_handler)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/tableschema/schema.py", line 266, in cast_row
    error_data=keyed_row)
  File "/home/didiez/anaconda3/lib/python3.7/site-packages/tableschema/helpers.py", line 90, in default_exc_handler
    raise exc
datapackage.exceptions.CastError: Row length 5 doesn't match fields count 6 for row "2"
```

---

Please preserve this line to notify @roll (lead of this repository)

**Comment by didiez** [comment]
Test to showcase the problem:
```python
# !/usr/bin/python
# -*- coding: utf-8 -*-

from goodtables import validate


DESCRIPTOR = {
    "resources": [
        {
            "name": "people",
            "data": [
                ["id", "name", "surname"],
                ["p1", "Tom", "Hanks"],
                ["p2", "Meryl", "Streep"]
            ],
            "schema": {
                "fields": [
                    {"name": "id", "type": "string"},
                    {"name": "name", "type": "string"},
                    {"name": "surname", "type": "string"},
                    {"name": "dob", "type": "date"}
                ]
            }
        },
        {
            "name": "oscars",
            "data": [
                ["person_id", "category", "year", "work"],
                ["p1", "Best Actor", 1994, "Philadelphia"],
                ["p1", "Best Actor", 1995, "Forrest Gump"],
                ["p2", "Best Supporting Actress", 1980, "Kramer vs. Kramer"],
                ["p2", "Best Actress", 1982, "Sophie's Choice"],
                ["p2", "Best Actress", 2012, "The Iron Lady"],
                ["p3", "Best Actor", 2019, "Joker"]
            ],
            "schema": {
                "fields": [
                    {"name": "person_id", "type": "string"},
                    {"name": "category", "type": "string"},
                    {"name": "year", "type": "year"},
                    {"name": "work", "type": "string"},
                ],
                "foreignKeys": [
                    {
                        'fields': 'persona_id',
                        'reference': {'resource': 'people', 'fields': 'id'}
                    }
                ]
            }
        }
    ]
}

print(validate(DESCRIPTOR, checks=['structure', 'schema', 'foreign-key']))

```

**Comment by roll** [comment]
Thanks,

Yea the problem that `datapackage` we use for reading a reference table uses strict validation. We need to think about how to extract this data without full validation

**Comment by roll** [comment]
FIXED in #365
